d3a_api_client/commands.py

```diff
@@ -0,0 +1,69 @@
+from d3a_api_client.enums import Commands, command_enum_to_command_name
+
+
+class ClientCommandBuffer:
+
+    def __init__(self, ):
+        self._commands_buffer = []
+
+    @property
+    def buffer_length(self):
+        return len(self._commands_buffer)
+
+    def offer_energy(self, area_uuid, energy, price):
+        self._add_to_buffer(area_uuid, Commands.OFFER, {"energy": energy, "price": price})
+        return self
```

Comment:
In order to avoid repeating `return self` in all these methods, you could `return self` at the end of `_add_to_buffer` method, and `return self._add_to_buffer()`.